relayer/chain.go

```diff
@@ -245,23 +249,37 @@ func (c *Chain) SendMsgs(msgs []sdk.Msg) (res *sdk.TxResponse, err error) {
 	// Build the transaction builder
 	txb, err := tx.BuildUnsignedTx(txf, msgs...)
 	if err != nil {
-		return nil, err
+		return nil, false, err
 	}
 
 	// Attach the signature to the transaction
 	err = tx.Sign(txf, c.Key, txb)
 	if err != nil {
-		return nil, err
+		return nil, false, err
 	}
 
 	// Generate the transaction bytes
 	txBytes, err := ctx.TxConfig.TxEncoder()(txb.GetTx())
 	if err != nil {
-		return nil, err
+		return nil, false, err
 	}
 
 	// Broadcast those bytes
-	return ctx.BroadcastTx(txBytes)
+	res, err := ctx.BroadcastTx(txBytes)
+	if err != nil {
+		return nil, false, err
+	}
+
+	// transaction was executed, log the success or failure using the tx response code
+	// NOTE: error is nil, logic should use the returned error to determine if the
+	// transaction was successfully executed.
+	if res.Code != 0 {
```

Comment:
I think it makes sense to cut down on the number of `Send` messages and just handle all the logic in one place

Reply:
This is much nicer. We could also add retries for account number in here too...